### PR TITLE
MRG: Use actual push button widgets for push buttons in Coreg UI

### PR DESCRIPTION
**Pull request by hoechenberger**
On my macOS machine:

|`main`|PR|
|-------|----|
| Screen Shot 2022-01-19 at 12 27 40 |  Screen Shot 2022-01-19 at 12 27 08 |

### CODE REVIEW

File: mne/viz/backends/_abstract.py [hunk 1]
```diff
@@ -532,7 +532,9 @@ def _dock_add_label(self, value, align=False, layout=None):
         pass
 
     @abstractmethod
-    def _dock_add_button(self, name, callback, tooltip=None, layout=None):
+    def _dock_add_button(
+        self, name, callback, style='pushbutton', tooltip=None, layout=None
```
Comment (larsoner) [inline mne/viz/backends/_abstract.py:536]: If you're going to add to the middle, let's be safe
```suggestion
        self, name, callback, *, style='pushbutton', tooltip=None, layout=None
```

File: mne/gui/_coreg.py [hunk 2]
```diff
@@ -1422,6 +1430,7 @@ def _configure_dock(self):
             callback=self._reset_fitting_parameters,
             tooltip="Reset all the fitting parameters to default value",
             layout=fitting_options_layout,
+            style='pushbutton',
```
Comment (larsoner) [inline mne/gui/_coreg.py:1433]: I would not add `style='pushbutton'` to all of these. I think `pushbutton` is a clear and obvious default for that function (and should have always been that way). Adding this param to all of these calls makes it less clear to me actually, in that at first it seems like you *are* somehow deviating from whatever the reasonable defaults are. (When reading through this code I thought, "why isn't the default `'pushbutton'` already?" just to read farther down that it actually is.)

Comment (hoechenberger) [reply mne/gui/_coreg.py:1433]: Very good point, I will adjust this accordingly.

**Comment by hoechenberger** [comment]
@larsoner I've implemented your suggestions, except for the kw-only parameters (see [my comment above](https://github.com/mne-tools/mne-python/pull/10220#discussion_r787957882))

File: mne/viz/backends/_abstract.py (hunk 1 above)
Comment (GuillaumeFavelier) [reply mne/viz/backends/_abstract.py:536]: Exactly what I meant 👍

**Comment by larsoner** [comment]
Thanks @hoechenberger !

**Pull request merged**